onadata/apps/logger/models/xform.py

```diff
@@ -534,13 +534,16 @@ def get_language(self, languages, language_index=0):
 
         return language
 
-    def get_label(self, abbreviated_xpath, elem=None, language_index=None):
+    def get_label(self, abbreviated_xpath, elem=None, language=None):
         elem = self.get_element(abbreviated_xpath) if elem is None else elem
 
         if elem:
             label = elem.label
 
-            if isinstance(label, dict):
+            if language and isinstance(label, dict):
+                label = label[language]
+
```

Comment:
Should we handle cases where a user passes a language that doesn't exist on the XForm to the `get_label` function ? _Seems the `self.get_language` function has a list of languages... or we could just use `label.get(language, <default>)`.._

Or is that something that's handled else where ?

Reply:
Such a situation has not been taken in to account; going to tweak this a bit